src/ess/dream/io/geant4.py

```diff
@@ -102,8 +102,10 @@ def _group(detectors: Dict[str, sc.DataArray]) -> Dict[str, sc.DataGroup]:
     def group(key: str, da: sc.DataArray) -> sc.DataArray:
         if key in ["high_resolution", "sans"]:
             # Only the HR and SANS detectors have sectors.
-            return da.group("sector", *elements)
-        res = da.group(*elements)
+            res = da.group("sector", *elements)
+        else:
+            res = da.group(*elements)
+        res.coords['position'] = res.bins.coords.pop('position').bins.mean()
```

Comment:
Did you check whether this changes the final result?

Reply:
I was hoping there are tests for that...?

Reply:
There are no tests for the complete $I(d)$. We only test a subset of properties of the output.